app/components/post_draft/send_handler/__tests__/send_handler.test.tsx

```diff
@@ -0,0 +1,198 @@
+    });
+
+    it('should render DraftInput when not from draft view', () => {
+        const wrapper = renderWithEverything(
+            <SendHandler {...baseProps}/>, {database},
+        );
+        expect(wrapper.getByTestId('test_send_handler')).toBeTruthy();
+    });
+
+    it('should render SendDraft when from draft view', () => {
+        const props = {
+            ...baseProps,
+            isFromDraftView: true,
+            draftType: DRAFT_TYPE_DRAFT as DraftType,
+        };
+        const wrapper = renderWithEverything(
+            <SendHandler {...props}/>, {database},
+        );
+        expect(wrapper.getByTestId('send_draft_button')).toBeTruthy();
+        expect(wrapper.getByText('Send draft')).toBeTruthy();
+    });
+
+    it('should render Send text when draft type is scheduled', () => {
+        const props = {
+            ...baseProps,
+            isFromDraftView: true,
+            draftType: DRAFT_TYPE_SCHEDULED as DraftType,
+        };
+        const wrapper = renderWithEverything(
+            <SendHandler {...props}/>, {database},
+        );
+        expect(wrapper.getByTestId('send_draft_button')).toBeTruthy();
+        expect(wrapper.getByText('Send')).toBeTruthy();
+    });
+
+    it('should handle post priority updates', async () => {
+        const wrapper = renderWithEverything(
+            <SendHandler
+                {...baseProps}
+                canShowPostPriority={true}
+            />, {database},
+        );
+
+        const draftInput = wrapper.getByTestId('test_send_handler');
+        expect(draftInput).toBeTruthy();
+        expect(baseProps.postPriority.priority).toBe('urgent');
+        await waitFor(() => {
+            expect(wrapper.toJSON()).toBeTruthy();
+        });
+    });
+
+    it('should pass correct props to SendDraft when in draft view', () => {
+        const props = {
+            ...baseProps,
+            isFromDraftView: true,
+            draftType: DRAFT_TYPE_DRAFT as DraftType,
+            channelDisplayName: 'Test Channel',
+            draftReceiverUserName: 'test-user',
+            postId: 'test-post-id',
+        };
+
+        const wrapper = renderWithEverything(
+            <SendHandler {...props}/>, {database},
+        );
+
+        const sendDraftButton = wrapper.getByTestId('send_draft_button');
+        expect(sendDraftButton).toBeTruthy();
+    });
+
+    it('should handle send message action', async () => {
+        const mockClearDraft = jest.fn();
+        const props = {
+            ...baseProps,
+            value: 'test message',
+            clearDraft: mockClearDraft,
+        };
+
+        const wrapper = renderWithEverything(
+            <SendHandler {...props}/>, {database},
+        );
+
+        const draftInput = wrapper.getByTestId('test_send_handler');
+        expect(draftInput).toBeTruthy();
+
+        await waitFor(() => {
+            expect(wrapper.toJSON()).toBeTruthy();
+        });
```

Comment:
Not sure how this translates to `should handle send message action`.

Reply:
Updated the test.